include/dsn/c/api_layer1.h

```diff
@@ -785,6 +785,9 @@ extern DSN_API void          dsn_msg_write_next(
 /*! commit the write buffer after the message content is written with the real written size */
 extern DSN_API void          dsn_msg_write_commit(dsn_message_t msg, size_t size);
 
+/*! append a buffer to the message, the "data" can be reinterpreted cast to (blob*)  */
+extern DSN_API void          dsn_msg_write_append(dsn_message_t msg, dsn_blob_t data);
```

Comment:
dsn_blob_t is not a POD or C structure, therefore we cannot put it in this C API. May use ```(dsn_message_t, void*, size_t)``` instead.